instrumentation/ktor/ktor-2-common/library/src/main/kotlin/io/opentelemetry/instrumentation/ktor/v2_0/common/AbstractKtorClientTelemetryBuilder.kt

```diff
@@ -78,8 +79,13 @@ abstract class AbstractKtorClientTelemetryBuilder(
     })
   }
 
+  @Deprecated("Use spanNameExtractor(UnaryOperator) instead", ReplaceWith("spanNameExtractor(spanNameExtractorTransformer as UnaryOperator<SpanNameExtractor<HttpRequestData>>)"))
   fun spanNameExtractor(spanNameExtractorTransformer: Function<SpanNameExtractor<HttpRequestData>, SpanNameExtractor<HttpRequestData>>) {
-    builder.setSpanNameExtractor(spanNameExtractorTransformer)
+    spanNameExtractor(spanNameExtractorTransformer as UnaryOperator<SpanNameExtractor<HttpRequestData>>)
```

Comment:
when you add
```kotlin
    prepare {
      val f = Function { t: SpanNameExtractor<HttpRequestData> -> t }
      this.spanNameExtractor(f)
    }
```
to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/55d6ae45dbb90877a5c218c06bd39b15c056ee2b/instrumentation/ktor/ktor-3.0/library/src/test/kotlin/io/opentelemetry/instrumentation/ktor/v3_0/KtorHttpClientTest.kt#L21
you'll get a `ClassCastException`. Apparently `spanNameExtractor(spanNameExtractorTransformer::apply)` works in kotlin too.